src/edu/washington/escience/myria/parallel/Server.java

```diff
@@ -1036,6 +1037,52 @@ public void importDataset(final RelationKey relationKey, final Schema schema, fi
   }
 
   /**
+   * @param relationKey the relationKey of the dataset to delete
+   * @return the status
+   * @throws DbException if there is an error
+   * @throws InterruptedException interrupted
+   */
+  public DatasetStatus deleteDataset(final RelationKey relationKey) throws DbException, InterruptedException {
+
+    /* Tombstone the relation */
+    try {
+      catalog.tombstoneRelationOnCatalog(relationKey);
+    } catch (CatalogException e) {
+      throw new DbException(e);
+    }
+
+    /* Delete from postgres at each worker by calling the DbDelete operator */
+    try {
+      Map<Integer, SubQueryPlan> workerPlans = new HashMap<>();
+      for (Integer workerId : getWorkersForRelation(relationKey, null)) {
+        workerPlans.put(workerId, new SubQueryPlan(new DbDelete(EmptyRelation.of(catalog.getSchema(relationKey)),
+            relationKey, null)));
+      }
+      ListenableFuture<Query> qf =
+          queryManager.submitQuery("delete " + relationKey.toString(), "delete " + relationKey.toString(),
+              "deleting from " + relationKey.toString(getDBMS()), new SubQueryPlan(new SinkRoot(new EOSSource())),
+              workerPlans);
+      try {
+        qf.get();
+      } catch (ExecutionException e) {
+        throw new DbException("Error executing query", e.getCause());
```

Comment:
If the DbDelete operator fails when the partition doesn't exist on a worker, don't we need to handle the relevant exception here to ensure the API call succeeds when it is retried after a partial failure (i.e., some but not all partitions were already deleted)?